server/store/sqlstore/migrate.go

```diff
@@ -0,0 +1,136 @@
+package sqlstore
+
+import (
+	"context"
+	"embed"
+	"fmt"
+
+	"github.com/mattermost/morph"
+	"github.com/mattermost/morph/drivers"
+	"github.com/mattermost/morph/drivers/postgres"
+	"github.com/mattermost/morph/sources/embedded"
+)
+
+const (
+	migrationRemoteIDRequiredVersion         = 12
+	migrationWhitelistedUsersRequiredVersion = 13
+)
+
+//go:embed migrations/*.sql
+var Assets embed.FS
+
+func (s *SQLStore) Migrate(remoteID string) error {
+	driver, err := postgres.WithInstance(s.db)
+	if err != nil {
+		return fmt.Errorf("cannot create postgres driver: %w", err)
+	}
+
+	assetsList, err := Assets.ReadDir("migrations")
+	if err != nil {
+		return fmt.Errorf("cannot read assets dir: %w", err)
+	}
+
+	assetsNames := make([]string, len(assetsList))
+	for i, entry := range assetsList {
+		assetsNames[i] = entry.Name()
+	}
+
+	migrationAssets := &embedded.AssetSource{
+		Names: assetsNames,
+		AssetFunc: func(name string) ([]byte, error) {
+			return Assets.ReadFile("migrations/" + name)
+		},
+	}
+
+	src, err := embedded.WithInstance(migrationAssets)
+	if err != nil {
+		return err
+	}
+
+	opts := []morph.EngineOption{
+		morph.WithLock("msteams-lock-key"),
+		morph.SetMigrationTableName("msteamssync_schema_migrations"),
+		morph.SetStatementTimeoutInSeconds(1000000),
+	}
+
+	s.api.LogDebug("Creating migration engine")
+	engine, err := morph.New(context.Background(), driver, src, opts...)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		s.api.LogDebug("Closing migration engine")
+		engine.Close()
+	}()
+
+	return s.runMigrationSequence(engine, driver, remoteID)
+}
+
+// runMigrationSequence executes all the migrations in order, both
+// plain SQL and data migrations.
+func (s *SQLStore) runMigrationSequence(engine *morph.Morph, driver drivers.Driver, remoteID string) error {
+	if mErr := s.ensureMigrationsAppliedUpToVersion(engine, driver, migrationRemoteIDRequiredVersion); mErr != nil {
+		return mErr
+	}
+
+	if remoteID != "" {
```

Comment:
As I recall, these are data migrations that require running code vs. just SQL. Is the idea to continue to append to this method as such migrations are required, or in the long run would we split apart the two concepts (schema vs data)? I realize that would require some finagling given the need to run some data migrations "between" others.

@jespino, is this some shared channels complexity would could eliminate if we required users adopting two way sync to "reset" somehow?